### ISSUE TITLE
watchRun issue with latest version

### ISSUE THREAD

**Issue by tborychowski** [sveltejs/svelte-loader#63]
I updated `svelte-loader` to the latest version
```json
{
    "svelte": "^2.5.0",
    "svelte-loader": "^2.9.0"
}
```

and now I'm getting this issue:

```
[Error running Gulp] Error: ./client/app/index.html
Module build failed: TypeError: Cannot read property 'watchRun' of undefined
    at new VirtualModulesPlugin (/Users/me/project1/node_modules/svelte-loader/lib/virtual.js:35:17)
    at Object.module.exports (/Users/me/project1/node_modules/svelte-loader/index.js:83:46)
 @ ./client/index.js 1:0-24
```

**Edit:** this seems to be an issue with the `svelte-loader` only. When I downgrade to v2.8.1 - everything works.

**Comment by perifer** [comment]
Looking at the referenced https://github.com/sysgears/webpack-virtual-modules they do this:

```js
    var watchRunHook = function(watcher, callback) {
        self._watcher = watcher.compiler || watcher;
        callback();
    }

    if (compiler.hooks) {
        compiler.hooks.watchRun.tapAsync('VirtualModulesPlugin', watchRunHook);
    } else {
        compiler.plugin("watch-run", watchRunHook);
    }
```

My _very_ limited understanding of this from reading through the commits of the above repo is that `compiler.hooks` is for older versions of Webpack.

**Comment by SimulatedGREG** [comment]
After testing @perifer's comment, I did some more research and found that `compiler.hooks` were introduced in `webpack@4.0.0`. Applying this change with `webpack@3.12.0` fixes the issue.

Given that not everybody may be able to updated to v4 just yet, I've made a PR to support older versions. https://github.com/sveltejs/svelte-loader/pull/68

**Comment by Rich-Harris** [comment]
Fixed in 2.10.1, thanks @SimulatedGREG!
